proto/stream_plan.proto

```diff
@@ -89,15 +92,15 @@ message HashJoinNode {
   // on-the-fly within the plan.
   // TODO: remove this in the future when we have a separate DeltaHashJoin node.
   bool is_delta_join = 5;
-  // Used for internal table states. Id of the left table.
-  uint32 left_table_id = 6;
-  // Used for internal table states. Id of the right table.
-  uint32 right_table_id = 7;
+  // Used for internal table states.
+  catalog.Table left_table = 6;
+  // Used for internal table states.
+  catalog.Table right_table = 7;
   repeated uint32 dist_key_l = 8;
   repeated uint32 dist_key_r = 9;
```

Comment:
maybe we can merge these fields. but delaying this after the executor really uses the catalog in proto is ok.